service/src/main/java/bio/terra/workspace/db/model/UniquenessCheckParameters.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.workspace.db.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * This class is returned from resource handlers to the ResourceDao. It describes the uniqueness
+ * check that the ResourceDao should do. At this time, only string compares are supported, since
+ * those are all that are in use. We can make this more complex if we need other datatypes.
+ */
+public class UniquenessCheckParameters {
```

Comment:
I'd call these attributes rather than parameters, to match usage elsewhere. They  only become parameters when you sql-ify them.